### PR TITLE
feat: using SnippetManager in the Service Details to create code snippet

### PR DESCRIPTION
**Pull request by axel7083**
### What does this PR do?

This PR is using the SnippetManager in the frontend to generate code snippet. 

### Screenshot / video of UI

https://github.com/projectatomic/ai-studio/assets/42176370/1dfc04d1-d728-43ef-876d-8b0f7485ef10

### What issues does this PR fix or reference?

Fixes https://github.com/projectatomic/ai-studio/issues/555 https://github.com/projectatomic/ai-studio/issues/403

### How to test this PR?

- [x] write unit tests

### CODE REVIEW

File: packages/frontend/src/pages/InferenceServerDetails.svelte [hunk 1]
```diff
@@ -74,6 +128,49 @@ $: service = $inferenceServers.find(server => server.container.containerId === c
...
+                  on:change="{onLanguageChange}"
+                  id="languages"
+                  class="border text-sm rounded-lg focus:ring-purple-500 focus:border-purple-500 block p-1 bg-charcoal-900 border-charcoal-900 placeholder-gray-700 text-white"
+                  name="languages">
+                  {#each $snippetLanguages as language}
+                    <option class="my-1" value="{language.key}">{language.label}</option>
+                  {/each}
+                </select>
+                {#if selectedVariant !== undefined}
+                  <select
```
Comment (jeffmaury) [inline packages/frontend/src/pages/InferenceServerDetails.svelte:151]: The select should be disable if the language has only one variant

**Review by jeffmaury — APPROVED** [review]
LGTM

File: packages/frontend/src/pages/InferenceServerDetails.svelte (hunk 1 above)
Comment (jeffmaury) [reply packages/frontend/src/pages/InferenceServerDetails.svelte:151]: Yes

**Comment by feloy** [comment]
When I try with curl, the path seems incorrect, as I get a 404. The path is `/` in the snippet, it seems it should be `/v1/chat/completions`

File: packages/frontend/src/pages/InferenceServerDetails.svelte (hunk 1 above)
Comment (axel7083) [reply packages/frontend/src/pages/InferenceServerDetails.svelte:151]: I would rather apply the suggestion of @jeffmaury as displaying `N/A` or `Not Applicable` could be weird, when we do have a name for the variant. Showing a hint, of the library used generally.

Here is why it look likes when it is disabled when only one variant.

image

image

image

**Comment by axel7083** [comment]
> When I try with curl, the path seems incorrect, as I get a 404. The path is `/` in the snippet, it seems it should be `/v1/chat/completions`

Oh nice catch @feloy, thanks for noticing. I fixed it

image

Here is what is resulting when executing the curl command

image

**Review by feloy — APPROVED** [review]
great!

**Pull request merged**